pkg/util/filesystem/permission_windows.go

```diff
@@ -22,6 +22,11 @@ type Permission struct {
 	systemSid        *windows.SID
 }
 
+// return the user dd-agent uid and gid on Unix, windows doesn't support unix socket credential
+func UserDDAgent() (found bool, usrID int, grpID int, err error) {
```

Comment:
The intent of this function is being mixed with how it is used. This function should return the UID and GID for the `dd-agent` user. It should not overload it with how it is being used for unix socket credentials. That logic should happen elsewhere.